internal/v1/server.go

```diff
@@ -758,6 +765,19 @@ func buildCustomizations(cust *Customizations) *composer.Customizations {
 		res.Filesystem = &fsc
 	}
 
+	if cust.Users != nil {
+		users := make([]composer.User, len(*cust.Users))
```

Comment:
I don't think this is correct, it will create a slice with the same length as `cust.Users` and then append to it. It means that the slice will have empty members at the beginning. See https://go.dev/tour/moretypes/13

It would be also great if we can cover this conversion process with a test, it would have actually uncovered this issue I think.